task_DEV-7

Zmitsier-Sakhonchyk opened the pull request.

Commit:
Add task_DEV-7

Commit:
Update Solution.sln

IgorKheidorov commented on task_DEV-7/Input.cs:
```diff
@@ -0,0 +1,53 @@
+            string[] triangleSides;
+            if (args.Length == 0 || indicator == true)
+            {
+
```
Codestyle

Commit:
Update Triangle.cs

Added triangle inequality check.

Commit:
Update Input.cs

Commit:
Update Triangle.cs

Commit:
Update TriangleAnalysisTool.cs

Commit:
Add TriangleSidesValidator.cs

Commit:
Update Input.cs

Commit:
Update TriangleAnalysisTool.cs

Changed try-catch location.

IgorKheidorov commented on task_DEV-7/Input.cs:
```diff
@@ -0,0 +1,96 @@
+                    validator.CheckValidity();
+                    repeat = false;
+                }
+                catch (FormatException e)
```
No need to catch each type of possible exception, the processing is similar to all, use only one catch

IgorKheidorov commented on task_DEV-7/Input.cs:
```diff
@@ -0,0 +1,96 @@
+
+        public string[] EnterTriangleSides()
+        {
+            string[] triangleSides = null;
```
No need to use local variable, enter values directly to args

IgorKheidorov commented on task_DEV-7/Triangle.cs:
```diff
@@ -0,0 +1,68 @@
+
+        public void DetermineType()
+        {
+            if ((Double.Parse(sides[0]) == Double.Parse(sides[1])) && 
```
What is it???? never compare doubles in such a way!!!! It can't work! Be more careful on lectures or study doubles comparison methods yourself.

IgorKheidorov commented on task_DEV-7/Triangle.cs:
```diff
@@ -0,0 +1,68 @@
+
+    class Triangle
+    {
+        private int type;
```
Why use int not enum????

IgorKheidorov reviewed (commented):
Re-work, principle errors

Commit:
Update Input.cs

Removed local variable.
Changed method EnterTriangleSides().
Removed unnecessary catch blocks.

Commit:
Update Triangle.cs

Changed field type (int to enum).
Changed double pair comparison method.
